src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -159,13 +159,19 @@ private[hyperspace] case class IndexSummary(
 
 private[hyperspace] object IndexSummary {
   def apply(spark: SparkSession, entry: IndexLogEntry): IndexSummary = {
+    var root = entry.content.root
+    var indexDirPath = entry.content.root.name
+    while (root.files.isEmpty) {
+      root = root.subDirs.head
+      indexDirPath += root.name + Path.SEPARATOR
```

Comment:
can we just use `Path` for concatenation?

Reply:
thanks @imback82 , done